spec/solver_spec.rb

```diff
@@ -0,0 +1,26 @@
+require './solver'
+require 'rspec'
+
+describe Solver do
+  before :each do
+    @solver = Solver.new
+  end
+
+  describe '#factorial' do
+    it 'successfully implements the mathematical factorial method' do
+      expect(@solver.factorial(5)).to eq 120
+    end
+  end
+
+  describe '#reverse' do
+    it 'reverse the input string' do
+      expect(@solver.reverse('keenan')).to eq 'naneek'
+    end
+  end
+
+  describe '#fizzbuzz' do
+    it 'returns constructed string' do
+      expect(@solver.fizzbuzz(15)).to eq '1, 2, fizz, 4, buzz, fizz, 7, 8, fizz, buzz, 11, fizz, 13, 14, fizzbuzz'
+    end
+  end
```

Comment:
- [ ] Good job writing the tests, however, according to the requirements below, kindly add the following tests for factorial method
- Negative value which raises an exception as an expectation
- Special case - when input is 0, the result should equal 1

- [ ] For the fizzbuzz method, kindly add the following tests
```ruby
expect(resolver.fizzbuzz(3)).to eql('fizz')  # N is divisible by 3
    ...
    expect(resolver.fizzbuzz(5)).to eql('buzz')  # N is divisible by 5
    ...
    expect(resolver.fizzbuzz(15)).to eql('fizzbuzz') # N is divisible by 3 and 5
    ...
    expect(resolver.fizzbuzz(7)).to eql('7')  # N is not divisible by 3 or 5
```

![Screenshot 2023-02-10 at 11 04 01 (2)](https://user-images.githubusercontent.com/64774936/218064003-e2be9dab-83fa-4a83-af91-563b8d32b63f.png)